src/main/java/org/symphonyoss/symphony/messageml/markdown/nodes/form/DatePickerNode.java

```diff
@@ -1,18 +1,32 @@
 package org.symphonyoss.symphony.messageml.markdown.nodes.form;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
 public class DatePickerNode extends FormElementNode {
   private final static String MARKDOWN = "Date Picker";
 
+  private String label;
+  private String tooltip;
   private String placeholder;
 
-  public DatePickerNode(String placeholder) {
+  public DatePickerNode(String label, String tooltip, String placeholder) {
     super(MARKDOWN, placeholder);
+    this.label = label;
+    this.tooltip = tooltip;
     this.placeholder = placeholder;
   }
 
   @Override
   public String getText() {
-    return (placeholder != null) ? String.format(":[%s]", placeholder) : "";
+    List<String> text = Arrays.asList(label, tooltip, placeholder)
```

Comment:
[optional comment]

- `Stream.of()` instead of `Arrays.asList(label, tooltip, placeholder).stream()`
- `.filter(e -> e != null && !e.isEmpty())` could be split in `.filter(Objects::nonNull).filter(e -> !e.isEmpty())`

Reply:
I agree @symphony-thibault but even better I refactored with a single stream one shot, no if-else, no second loop to build response...much more elegant and coincise